From 144f8964b19638ab3a2eee6ff0acf0024dda0e24 Mon Sep 17 00:00:00 2001
From: Zhipeng Gong <zhipeng.gong@intel.com>
Date: Mon, 12 Sep 2016 21:01:19 -0400
Subject: [PATCH 146/153] [VPG]: drm/i915: Set RING/GT frequency multiplier to 3:1

Change RING/GT ratio from 2:1 to 3:1, which can reduce the transitions
of RING between different frequencies, frequent transitions can affect
the performance.
---
 drivers/gpu/drm/i915/intel_pm.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_pm.c b/drivers/gpu/drm/i915/intel_pm.c
index dfd6ecf..2372070 100644
--- a/drivers/gpu/drm/i915/intel_pm.c
+++ b/drivers/gpu/drm/i915/intel_pm.c
@@ -5080,10 +5080,10 @@ static void __gen6_update_ring_freq(struct drm_device *dev)
 
 		if (IS_SKYLAKE(dev)) {
 			/*
-			 * ring_freq = 2 * GT. ring_freq is in 100MHz units
+			 * ring_freq = 3 * GT. ring_freq is in 100MHz units
 			 * No floor required for ring frequency on SKL.
 			 */
-			ring_freq = gpu_freq;
+			ring_freq = mult_frac(gpu_freq, 3, 2);
 		} else if (INTEL_INFO(dev)->gen >= 8) {
 			/* max(2 * GT, DDR). NB: GT is 50MHz units */
 			ring_freq = max(min_ring_freq, gpu_freq);
-- 
1.7.1

